integration-tests/rest-openapi/src/main/java/org/apache/camel/quarkus/component/rest/openapi/it/RestOpenapiResource.java

```diff
@@ -27,15 +30,54 @@
 import org.apache.camel.ProducerTemplate;
 
 @Path("/rest-openapi")
+@ApplicationScoped
 public class RestOpenapiResource {
     @Inject
     ProducerTemplate producerTemplate;
 
-    @Path("/fruits/list")
+    @Path("/fruits/list/json")
     @Produces(MediaType.APPLICATION_JSON)
     @GET
     public Response invokeListFruitsOperation(@QueryParam("port") int port) {
-        String response = producerTemplate.requestBodyAndHeader("direct:start", null, "test-port", port, String.class);
+        return invokeListFruitsOperation("start-web-json", port);
+    }
+
+    @Path("/fruits/list/yaml")
+    @Produces(MediaType.APPLICATION_JSON)
+    @GET
+    public Response invokeListFruitsOperationYaml(@QueryParam("port") int port) {
+        return invokeListFruitsOperation("start-web-yaml", port);
+    }
+
+    @Path("/fruits/list/file")
+    @Produces(MediaType.APPLICATION_JSON)
+    @GET
+    public Response invokeListFruitsOperationFile(@QueryParam("port") int port) {
+        Response response = invokeListFruitsOperation("start-file", port);
+        File openApiFile = new File("output", "openapi.json");
```

Comment:
Is it really test logic ? I wonder if this be better placed in a `@BeforeAll` method. So that we know, the file is never present before the test.

Reply:
I moved that logic to the test, which also simplified the route for that test.